javaslang/src-gen/main/java/javaslang/CheckedFunction0.java

```diff
@@ -75,8 +75,8 @@
      * @return a function that applies arguments to the given {@code partialFunction} and returns {@code Some(result)}
      *         if the function is defined for the given arguments, and {@code None} otherwise.
      */
-    static <R> Function0<Option<R>> lift(CheckedFunction0<R> partialFunction) {
-        return () -> Try.of(partialFunction::apply).getOption();
+    static <R> Function0<Option<R>> lift(CheckedFunction0<? extends R> partialFunction) {
+        return () -> Try.of(of(partialFunction)::apply).getOption();
```

Comment:
We can replace the method call `of(partialFunction)` by providing a type hint `<R>`:

```java
return () -> Try.<R> of(partialFunction::apply).getOption();
```